### PR TITLE
Added the "trace-whitelist" flag to run.

### PR DESCRIPTION
**Pull request by gaaclarke**
## Description

Added "trace-whitelsit" flag to the run command to match the engine change: https://github.com/flutter/engine/pull/17108

## Related Issues

https://github.com/flutter/flutter/issues/40593

## Tests

yes

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/run.dart [hunk 1]
```diff
@@ -102,6 +102,10 @@ class RunCommand extends RunCommandBase {
         help: 'Enable tracing of Skia code. This is useful when debugging '
               'the GPU thread. By default, Flutter will not log skia code.',
       )
+      ..addOption('trace-whitelist',
+        help: 'Filters out all trace events except those that are specified in '
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/commands/run.dart]: Add a `valueHelp` with some examples of parameters that could be passed

**Comment by jonahwilliams** [comment]
Testssss 🐍

File: packages/flutter_tools/lib/src/commands/run.dart (hunk 1 above)
Comment (gaaclarke) [reply packages/flutter_tools/lib/src/commands/run.dart]: done

**Comment by gaaclarke** [comment]
> Testssss 🐍

I tried Jonah.  The tests are so incomplete and esoteric it's unreasonable to ask anyone to edit it.  Here is what I tried to do for android:  https://gist.github.com/gaaclarke/dd7e4b79c1d67aa770a2d433e2e4d561

There are no preexisting tests for all the flags.  I tried to tweak one that we do have for `cacheSkSL` but that one only works with debugging turned off.  I need to have debugging turned on because of the DebuggingOptions factory functions.  When you turn debugging on all these other subprocesses try to get launched behind the scene.  I could add another constructor for DebuggingOptions, but then i have to duplicate all the default values and we have to maintain that just for this test.

**Comment by jonahwilliams** [comment]
Well, if everyone had that attitude they wouldn't get any better! We need our tooling to have tests too, otherwise things will break as they get updated or refactored. Please give it a shot, or add what you've got so far to the review and I can provide guidance.

**Comment by gaaclarke** [comment]
> Well, if everyone had that attitude they wouldn't get any better! We need our tooling to have tests too, otherwise things will break as they get updated or refactored. Please give it a shot, or add what you've got so far to the review and I can provide guidance.

The other option is that we make an investment have people with the expertise to expand coverage to provide good examples and lower the bar to contribute.

Could you help with the test I wrote?  I have it in the gist in the previous comment.  When I switch to `DebuggingOptions.enabled` I get errors when the tool tries to run a subprocess and check its output.  I'm guessing there might be something we can do with `mockProcessManager` but I have no idea.

**Comment by gaaclarke** [comment]
@jonahwilliams Here is the stacktrace:
```
  NoSuchMethodError: The getter 'exitCode' was called on null.
  Receiver: null
  Tried calling: exitCode
  dart:core                                                      Object.noSuchMethod
  package:flutter_tools/src/base/process.dart 154:37             RunResult.exitCode
  package:flutter_tools/src/base/process.dart 424:47             _DefaultProcessUtils.runSync
  package:flutter_tools/src/android/android_device.dart 243:25   AndroidDevice.runAdbCheckedSync
  package:flutter_tools/src/android/android_device.dart 699:16   AndroidDevice.lastLogcatTimestamp
  package:flutter_tools/src/android/android_device.dart 1051:16  AdbLogReader.createLogReader
  ===== asynchronous gap ===========================
  dart:async                                                     _asyncErrorWrapperHelper
  test/src/context.dart 142:30                                   testUsingContext.<fn>.<fn>.<fn>.<fn>
  dart:async                                                     runZoned
  test/src/context.dart 140:18                                   testUsingContext.<fn>.<fn>.<fn>
  package:flutter_tools/src/base/context.dart 150:29             AppContext.run.<fn>
  dart:async                                                     runZoned
  package:flutter_tools/src/base/context.dart 149:18             AppContext.run
  test/src/context.dart 106:22                                   testUsingContext.<fn>.<fn>
  package:flutter_tools/src/context_runner.dart 69:18            runInContext.runnerWrapper
  ===== asynchronous gap ===========================
  dart:async                                                     _asyncThenWrapperHelper
  package:flutter_tools/src/context_runner.dart                  runInContext.runnerWrapper
  package:flutter_tools/src/base/context.dart 150:29             AppContext.run.<fn>
  dart:async                                                     runZoned
  package:flutter_tools/src/base/context.dart 149:18             AppContext.run
  package:flutter_tools/src/context_runner.dart 72:24            runInContext
  test/src/context.dart 105:11                                   testUsingContext.<fn>
```

Unfortunately it isn't very helpful because of the loss of information as a result of asynchronous calls and generic wrappers around subprocesses.

**Comment by jonahwilliams** [comment]
The reason for that is that the "enabled" path goes through a lot more logic that is mocked out. It would be easier to add the trace whitelist to DebuggingOptions.disabled also and keep using that

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Pull request merged**